src/Umbraco.Web/Editors/MediaController.cs

```diff
@@ -278,6 +281,31 @@ public IEnumerable<MediaItemDisplay> GetByIds([FromUri]int[] ids)
         }
 
         /// <summary>
+        /// Searches media and returns a paged result
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="pageNumber"></param>
+        /// <param name="pageSize"></param>
+        /// <param name="searchFrom"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [FilterAllowedOutgoingMedia(typeof(IEnumerable<ContentItemBasic<ContentPropertyBasic, IMedia>>), "Items")]

```

Comment:
As emil says we need some extra data to render the media grid: thumbnail, width, height etc. When this is said I think we have a bigger problem with the media picker because we use the media controller a couple of times in there: for getChildren, addFolder, upload media. The result right now is that you get logged out, if you don’t have acces to the media section, when you use the media picker. So we should also consider if we need to limit the functionality of the media picker if you don't have access to the media section.